homeassistant/components/supla/__init__.py

```diff
@@ -0,0 +1,158 @@
+"""Support for the Supla devices."""
+
+import logging
+from typing import Optional
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_ACCESS_TOKEN
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.discovery import load_platform
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pysupla==0.0.2']
+
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = 'supla'
+
+CONF_SERVER = 'server'
+CONF_SERVERS = 'servers'
+
+SUPLA_FUNCTION_HA_CMP_MAP = {
+    'CONTROLLINGTHEROLLERSHUTTER': 'cover'
+}
+SUPLA_CHANNELS = 'supla_channels'
+SUPLA_SERVERS = 'supla_servers'
+
+
+SERVER_CONFIG = vol.Schema({
+    vol.Required(CONF_SERVER): cv.string,
+    vol.Required(CONF_ACCESS_TOKEN): cv.string
+}, extra=vol.ALLOW_EXTRA)
```

Comment:
Don't allow extra.

Reply:
Hmm, it seems to be required to workaround the following error:
`
2019-04-09 22:20:51 ERROR (MainThread) [homeassistant.config] Invalid config for [supla]: [homeassistant] is an invalid option for [supla]. Check: supla->homeassistant. (See /home/mwegrzynek/Projekty/home-assistant/../home-assistant-config/configuration.yaml, line 70). Please check the docs at https://home-assistant.io/components/supla/
`
I checked few hub-like components (for example alexa, fibaro) and they all use ALLOW_EXTRA.
Can you provide me an example I can base my fix on?

Reply:
We should only allow extra in the outermost schema dict, `CONFIG_SCHEMA`. The `SERVER_CONFIG` will be nested inside `CONFIG_SCHEMA`.